src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
```

Comment:
> I don't know the situation in production, but in most UTs the name is a full qualified one. And thats why I put the "full package test" at first in my previous commit.

This might be because of https://github.com/checkstyle/checkstyle/blob/b4a337a599e853cdc44b19a76da28fcfab2956cd/src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java#L54 where we call `clazz.getName()` instead of `getSimpleName`.
I don't know of a reason for this nor can I see one.